fix(core): prevent concurrent downloads clobbering by using unique temp dirs

emyjamalian opened the pull request:
- This change ensures concurrent 'npx playwright install' processes do not clobber each other by downloading to a unique temporary directory before finalizing.
- Adds unique temp dir per download in packages/playwright-core/src/server/registry/browserFetcher.ts.
- Adds installation test tests/installation/concurrent-download.spec.ts validating concurrent installs.
- Test assertions avoid depending on stdout formatting.
- Fixes https://github.com/microsoft/playwright/issues/32179

emyjamalian committed:
fix(core): prevent concurrent downloads clobbering by using unique temp dirs

Fixes #<please-link-issue-if-any>. Adds temporary unique download directory to avoid race conditions during concurrent installs. Updates installation test to be resilient to stdout formatting.

emyjamalian commented:
@microsoft-github-policy-service agree

emyjamalian commented:
> @emyjamalian please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

@microsoft-github-policy-service agree

emyjamalian committed:
chore(lint): remove trailing spaces in browserFetcher and installation test

emyjamalian committed:
fix(core): download to unique temp dir and cleanup via removeFolders

Use removeFolders() for temp dir cleanup and restore ProgressBar type. This prevents concurrent installs from clobbering each other.

emyjamalian committed:
test(installation): fix trailing whitespace and duplicate use() in concurrent-download test

yury-s reviewed (commented):
The new test is failing on Windows.

emyjamalian committed:
test(installation): use separate browser paths for concurrent installs to prevent conflicts

emyjamalian committed:
test(installation): install to separate paths and verify aggregated components

emyjamalian committed:
test(installation): remove unused import from concurrent-download test

yury-s commented on tests/installation/concurrent-download.spec.ts:
```diff
@@ -0,0 +1,46 @@
+      }
+    })
+  ));
+  // Aggregate installed software across individual browser paths and verify expected components exist
```
Why aggregate if you can simply check it in each directory? Just move the `installed` set and all 3 checks inside the loop - each installation should have all 3 binaries present.

emyjamalian committed:
test: improve concurrent download test to verify each installation individually

- Move browser verification inside the Promise.all loop
- Each installation is now checked independently for required binaries
- Eliminates aggregation complexity and provides better test isolation
- Makes failures more specific and easier to debug

emyjamalian committed:
fix: remove trailing spaces in concurrent-download test
